### PR TITLE
[SYCLomatic] Refine migration of texture APIs with bindless image.

### PR DESCRIPTION
**Pull request by tangjj11**
Signed-off-by: Tang, Jiajun jiajun.tang@intel.com

### CODE REVIEW

**Review by ziranzha — APPROVED** [review]

File: clang/runtime/dpct-rt/include/dpct/image.hpp [hunk 1]
```diff
@@ -840,6 +840,658 @@ static inline image_wrapper_base *create_image_wrapper(image_data data,
   return nullptr;
 }
 
+#ifdef SYCL_EXT_ONEAPI_BINDLESS_IMAGES
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/image.hpp]: Please put the new added helper code  bindless_images.hpp, and using experimental

File: clang/runtime/dpct-rt/include/dpct/image.hpp [hunk 2]
```diff
@@ -840,6 +840,658 @@ static inline image_wrapper_base *create_image_wrapper(image_data data,
...
+inline std::pair<image_data, sampling_info> &get_img_info_map(
+    const sycl::ext::oneapi::experimental::sampled_image_handle handle) {
+  static std::map<sycl::ext::oneapi::experimental::sampled_image_handle,
+                  std::pair<image_data, sampling_info>,
+                  sampled_image_handle_compare>
+      img_info_map;
+  return img_info_map[handle];
+}
+
+inline image_mem_p &get_img_mem_map(
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/image.hpp]: for public interface let's use "sycl::ext::oneapi::experimental::image_mem *".

File: clang/runtime/dpct-rt/include/dpct/image.hpp [hunk 3]
```diff
@@ -840,6 +840,658 @@ static inline image_wrapper_base *create_image_wrapper(image_data data,
   return nullptr;
 }
 
+#ifdef SYCL_EXT_ONEAPI_BINDLESS_IMAGES
+
+using image_mem_p = sycl::ext::oneapi::experimental::image_mem *;
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/image.hpp]:
```suggestion
using image_mem_ptr = sycl::ext::oneapi::experimental::image_mem *;
```

File: clang/runtime/dpct-rt/include/dpct/image.hpp (hunk 1 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/image.hpp]: fixed

File: clang/runtime/dpct-rt/include/dpct/image.hpp (hunk 2 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/image.hpp]: fixed

File: clang/runtime/dpct-rt/include/dpct/image.hpp (hunk 3 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/image.hpp]: fixed

File: clang/runtime/dpct-rt/include/dpct/dpct.hpp [hunk 4]
```diff
@@ -18,8 +18,8 @@ template <class... Args> class dpct_kernel_name;
 template <int Arg> class dpct_kernel_scalar;
 
 #include "atomic.hpp"
+#include "bindless_images.hpp"
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/dpct.hpp]: Please keep image.hpp as the top level header file, and bindless_images.hpp as second  level.

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp [hunk 5]
```diff
@@ -0,0 +1,688 @@
...
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef __DPCT_BINDLESS_IMAGE_HPP__
+#define __DPCT_BINDLESS_IMAGE_HPP__
+
+#include <sycl/sycl.hpp>
+
+#include "image.hpp"
```
Comment (zhimingwang36) [inline clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]: Please have image.hpp to include bindless_images.hpp.

File: clang/runtime/dpct-rt/include/dpct/dpct.hpp (hunk 4 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/dpct.hpp]: fixed

File: clang/runtime/dpct-rt/include/dpct/bindless_images.hpp (hunk 5 above)
Comment (tangjj11) [reply clang/runtime/dpct-rt/include/dpct/bindless_images.hpp]: fixed

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**